nodejs/aws-infra/examples/ec2/index.ts

```diff
@@ -288,11 +288,10 @@ const api = new aws.apigateway.x.API("examples-containers", {
             policies: [...x.ecs.TaskDefinition.defaultTaskRolePolicyARNs()],
             callback: async (req) => {
                 try {
-                    const c = cluster;
-                    await helloTask.run({ cluster: c });
+                    const result = await helloTask.run({ cluster }).promise();
```

Comment:
@CyrusNajmabadi Curious what you think about this `.promise()`?  I tried to keep this faithful to the API design of `aws-sdk`, which I think is the right call, but it does mean a very specific and unusual API design which may not be obvious given this is exposed apart from the `aws-sdk` itself.

Reply:
I'm pretty biased and i think that async/await has 'won the war'.  So my preference would be to just have the promisified form by default.  I don't think anyone is going to really be harmed by the *pulumi* layer being opinionated in that fashion.  

But i also don't feel super strongly about this.  i get the attractiveness of having the signatures be very close.  But it just feels like Promises are the better solution all the time.  

1. you can use async/await if you want.
2. if callback-style is preferable for you, you can always manually `.then` things. 

`.promise()` always feels like the way to deal with moving an older api forward.  I'd prefer to avoid it on the new apis.